workflow/annovar.wdl

```diff
@@ -0,0 +1,148 @@
+# Medical Research.                                               #
+#                                                                 #
+# Author: Michael Geaghan (micgea)                                #
+# Created: 2023/04/04                                             #
+# =============================================================== #
+
+struct AnnotationRuntime {
+    Int max_retries
+    Int preemptible
+    Int boot_disk_size
+}
+
+workflow Annovar {
+    input {
+        # input vcf
+        File input_vcf
+        # annovar settings
+        Int annovar_mem_mb = 4000
+        Int annovar_disk = 100
+        Int annovar_tmp_disk = 200
+        Int annovar_cpu = 1
+        String annovar_docker = "australia-southeast1-docker.pkg.dev/pb-dev-312200/somvar-images/perl@sha256:1f35086e2ff48dace3b3edeaa2ad1faf1e44c0612e00f00ea0fc1830b576a261"  # :5.34.0
+        File annovar_archive
+        String ref_name = "hg38"
+        String annovar_protocols = "refGene"
+        String annovar_operations = "g"
+        # runtime parameters
+        Int? preemptible
+        Int? max_retries
+        Int boot_disk_size = 12
+    }
+
+    Int preemptible_or_default = select_first([preemptible, 2])
+    Int max_retries_or_default = select_first([max_retries, 2])
+
+    AnnotationRuntime standard_runtime = {
+        "max_retries": max_retries_or_default,
+        "preemptible": preemptible_or_default,
+        "boot_disk_size": boot_disk_size
+    }
+
+    String sample_id = basename(basename(input_vcf, ".gz"), ".vcf")
+    call Annovar_task {
+        input:
+            vcf_input = input_vcf,
+            sample_id = sample_id,
+            mem_mb = annovar_mem_mb,
+            annovar_disk_space = annovar_disk,
+            annovar_tmp_disk_space = annovar_tmp_disk,
+            cpu = annovar_cpu,
+            annovar_docker = annovar_docker,
+            annovar_archive = annovar_archive,
+            ref_name = ref_name,
+            annovar_protocols = annovar_protocols,
+            annovar_operations = annovar_operations,
+            runtime_params = standard_runtime
+    }
+
+    output {
+        File out_annovar_vcf = Annovar_task.annovar_output_file_vcf
+        File out_annovar_table = Annovar_task.annovar_output_file_table
+    }
+}
+
+task Annovar_task {
+    input {
+      File vcf_input
+      String sample_id
+
+      Int mem_mb = 4000
+      Int annovar_disk_space = 100
+      Int annovar_tmp_disk_space = 200
+      Int cpu = 1
+      String annovar_docker
+      File annovar_archive
+
+      String label = "annovar_out"
+
+      String ref_name = "hg38"
+      String annovar_protocols = "refGene"
+      String annovar_operations = "g"
+      String annovar_additional_arguments = ""
+
+      AnnotationRuntime runtime_params
+    }
+
+    String file_prefix = sample_id + "." + label
+
+    String tmp_dir = "/tmp"
+
+    command {
+      set -euo pipefail
+
+      cd ~{tmp_dir}
+
+      tar -xzvf ~{annovar_archive}
```

Comment:
A nitpick, but this could make it hard to determine which exact version of scripts for provenance. Obviously it makes it a lot easier to tweak scripts, and your analysis is a bit different to ours, but this _technically_ is like an injection attack, effectively allowing unreviewed code to run.